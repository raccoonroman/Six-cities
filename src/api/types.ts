import { OfferCategory, Location } from '@/types';

export interface AuthInfoRaw {
  avatar_url: string;
  email: string;
  id: number;
  is_pro: boolean;
  name: string;
}

export interface User {
  email: string;
  password: string;
}

export interface OfferRaw {
  id: number;
  title: string;
  preview_image: string;
  price: number;
  rating: number;
  type: OfferCategory;
  bedrooms: number;
  max_adults: number;
  is_favorite: boolean;
  is_premium: boolean;
  location: Location;
  city: {
    name: string;
    location: Location;
  };
  description: string;
  goods: string[];
  host: {
    avatar_url: string;
    id: number;
    is_pro: boolean;
    name: string;
  };
  images: string[];
}

export interface CommentRaw {
  id: number;
  comment: string;
  date: string;
  rating: number;
  user: {
    avatar_url: string;
    id: number;
    is_pro: boolean;
    name: string;
  };
}

export interface CommentPost {
  comment: string;
  rating: number;
}
