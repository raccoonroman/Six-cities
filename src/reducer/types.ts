import { OfferRaw } from '@/api/types';
import { ActionType } from '@/const';

interface LoadOffersAction {
  type: typeof ActionType.LOAD_OFFERS
  payload: OfferRaw[]
}

interface UpdateOfferAction {
  type: typeof ActionType.UPDATE_OFFER
  payload: OfferRaw
}

interface SetEmailAction {
  type: typeof ActionType.SET_EMAIL
  payload: string
}

export type offersAction = LoadOffersAction | UpdateOfferAction;
export type userDataAction = SetEmailAction;
