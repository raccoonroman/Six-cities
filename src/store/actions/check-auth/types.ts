enum CheckAuth {
  PENDING = 'CHECK_AUTH/PENDING',
  RESOLVE = 'CHECK_AUTH/RESOLVE',
  REJECT = 'CHECK_AUTH/REJECT',
}

export default CheckAuth;
