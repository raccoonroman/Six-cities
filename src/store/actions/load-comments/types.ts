enum LoadComments {
  PENDING = 'LOAD_COMMENTS/PENDING',
  RESOLVE = 'LOAD_COMMENTS/RESOLVE',
  REJECT = 'LOAD_COMMENTS/REJECT',
}

export default LoadComments;
