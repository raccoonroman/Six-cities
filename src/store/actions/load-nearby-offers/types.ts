enum LoadNearbyOffers {
  PENDING = 'LOAD_NEARBY_OFFERS/PENDING',
  RESOLVE = 'LOAD_NEARBY_OFFERS/RESOLVE',
  REJECT = 'LOAD_NEARBY_OFFERS/REJECT',
}

export default LoadNearbyOffers;
