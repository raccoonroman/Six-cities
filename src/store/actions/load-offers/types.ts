enum LoadOffers {
  PENDING = 'LOAD_OFFERS/PENDING',
  RESOLVE = 'LOAD_OFFERS/RESOLVE',
  REJECT = 'LOAD_OFFERS/REJECT',
}

export default LoadOffers;
