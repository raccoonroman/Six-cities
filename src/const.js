export const DEFAULT_CITIES = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];

export const SortType = {
  POPULAR: `Popular`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};

export const CardType = {
  CITY: `city`,
  NEAR: `near`,
  FAVORITE: `favorite`,
};

export const ActionType = {
  LOAD_OFFERS: `LOAD_OFFERS`,
  UPDATE_OFFER: `UPDATE_OFFER`,
  SET_CITY: `SET_CITY`,
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SET_EMAIL: `SET_EMAIL`,
  LOAD_COMMENTS: `LOAD_COMMENTS`,
  LOAD_NEARBY_OFFERS: `LOAD_NEARBY_OFFERS`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER: `/offer`,
};
