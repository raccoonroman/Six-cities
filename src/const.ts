export const DEFAULT_CITIES: string[] = ['Paris', 'Cologne', 'Brussels', 'Amsterdam', 'Hamburg', 'Dusseldorf'];

export enum SortType {
  POPULAR = 'Popular',
  PRICE_LOW_TO_HIGH = 'Price: low to high',
  PRICE_HIGH_TO_LOW = 'Price: high to low',
  TOP_RATED_FIRST = 'Top rated first',
}

export enum CardType {
  CITY = 'city',
  NEAR = 'near',
  FAVORITE = 'favorite',
}

export enum ActionType {
  LOAD_OFFERS = 'LOAD_OFFERS',
  UPDATE_OFFER = 'UPDATE_OFFER',
  SET_CITY = 'SET_CITY',
  REQUIRED_AUTHORIZATION = 'REQUIRED_AUTHORIZATION',
  SET_EMAIL = 'SET_EMAIL',
  LOAD_COMMENTS = 'LOAD_COMMENTS',
  LOAD_NEARBY_OFFERS = 'LOAD_NEARBY_OFFERS',
}

export enum AuthorizationStatus {
  AUTH = 'AUTH',
  NO_AUTH = 'NO_AUTH',
}

export enum AppRoute {
  ROOT = '/',
  LOGIN = '/login',
  FAVORITES = '/favorites',
  OFFER = '/offer',
}

export enum MapType {
  HOVERED_ACTIVE_OFFER = 'HOVERED_ACTIVE_OFFER',
  STATIC_ACTIVE_OFFER = 'STATIC_ACTIVE_OFFER',
}
