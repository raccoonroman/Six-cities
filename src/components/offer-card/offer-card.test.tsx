import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {BrowserRouter, Route} from 'react-router-dom';
import {AuthorizationStatus, CardType, AppRoute} from '../../const';
import {Offer} from '../../types';
import OfferCard from './offer-card';


const mockStore = configureStore([]);

const offer: Offer = {
  id: 100502,
  title: `Excepteur sint occaecat cupidatat non proident`,
  previewImage: `https://i.picsum.photos/id/24/260/200.jpg`,
  price: 13,
  rating: 4.9,
  type: `house`,
  bedrooms: 3,
  maxAdults: 3,
  isFavorite: true,
  isPremium: false,
  location: {
    latitude: 52.359160,
    longitude: 4.849366,
    zoom: 12,
  },
  city: {
    name: `Vinnytsia`,
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 13,
    },
  },
};

const handleCardHover = () => {
  // do nothing.
};


it(`Should offer card render correctly`, () => {
  const store = mockStore({
    authorization: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
    },
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <Route exact path={AppRoute.ROOT} render={({history}) => (
              <OfferCard
                history={history}
                cardType={CardType.NEAR}
                offer={offer}
                onCardHover={handleCardHover}
              />
            )} />
          </BrowserRouter>
        </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
