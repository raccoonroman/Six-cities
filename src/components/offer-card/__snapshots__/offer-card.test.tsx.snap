// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should offer card render correctly 1`] = `
<article
  className="near-places__card place-card"
>
  <div
    className="near-places__image-wrapper place-card__image-wrapper"
  >
    <a
      href="/offer/100502"
      onClick={[Function]}
    >
      <img
        alt="Excepteur sint occaecat cupidatat non proident"
        className="place-card__image"
        height="200"
        src="https://i.picsum.photos/id/24/260/200.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          13
        </b>
        <span
          className="place-card__price-text"
        >
           / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button place-card__bookmark-button--active"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "100%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer/100502"
        onClick={[Function]}
      >
        Excepteur sint occaecat cupidatat non proident
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      house
    </p>
  </div>
</article>
`;
